### PR TITLE
Encounters makes you enconter a combat

### PR DESCRIPTION
**Pull request by x183**

### CODE REVIEW

**Review by x183 — COMMENTED** [review]
looks good to me, but I wrote parts of it :sweat_smile:

File: documents/com.g16.feyrune.OverworldModel.yuml [hunk 1]
```diff
@@ -7,7 +7,9 @@
 [MapParser] -.-> [Map]
 [Map] ++1-n> [Tile | isColition():boolean]
 [OverworldModel] -> [EncounterHandler|+getEncounter():Encounter;]
-[EncounterHandler] -.-> [CreatureFactory | +createCreature():ICreature]
+[EncounterHandler] -.-> [CreatureFactory |  +createCreatureList(List<Pair<String, Integer>>, int):List<ICreature>; +createCreature(String,int,int,int,int)]
```
Comment (lutches) [inline documents/com.g16.feyrune.OverworldModel.yuml:10]: attackFactory!!!
it an abilityFactory in the code. do not confuse

**Review by lutches — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/g16/feyrune/model/Model.java [hunk 2]
```diff
@@ -20,12 +20,12 @@ public class Model {
     public Model() {
         player = new Player("Player", new Point(20, 5)); //TODO: Should probably have a method to get the initial player position from the map
         this.overworldModel = new OverworldModel(player);
+        this.overworldModel.addObserver(this);
         this.observers = new ArrayList<>();
         stateHandler = new StateHandler(ModelState.WORLD);
     }
 
     public void update() {
```
Comment (lutches) [inline core/src/main/java/com/g16/feyrune/model/Model.java:28]: why is this function empty?

**Review by lutches — APPROVED** [review]
even though I believe the uml should have been changed in another pull request first I approve of this pull request.

File: core/src/main/java/com/g16/feyrune/model/overworld/map/Tile.java [hunk 3]
```diff
@@ -16,4 +16,7 @@ public boolean isCollision() {
     public boolean canEncounter() {
         return canEncounter;
     }
+    public void removeEncounter(){
+        canEncounter = false;
+    }
```
Comment (Krysset) [inline core/src/main/java/com/g16/feyrune/model/overworld/map/Tile.java:21]: When do we need to disable encounter on a tile after creating it?

File: core/src/main/java/com/g16/feyrune/model/Model.java (hunk 2 above)
Comment (Krysset) [reply core/src/main/java/com/g16/feyrune/model/Model.java:28]: It should update the model of the current state

**Pull request merged**